src/interactions/panZoomInteraction.ts

```diff
@@ -20,56 +29,143 @@ export module Interaction {
+
+      var oldAvgX = (leftX + rightX) / 2;
+      var oldAvgY = (bottomY + topY) / 2;
+      var oldDiffX = rightX - leftX;
+      var oldDiffY = bottomY - topY;
+
+      ids.forEach((id) => {
+        var translatedP = this._translateToComponentSpace(idToPoint[id]);
+        if (this._isInsideComponent(translatedP)) {
+          this._touchIds.set(id.toString(), translatedP);
+        }
+      });
+
+      var newPoints = this._touchIds.values();
+      var newFirstTouchPoint = newPoints[0];
+      var newSecondTouchPoint = newPoints[1];
+
+      var newLeftX = Math.min(newFirstTouchPoint.x, newSecondTouchPoint.x);
+      var newRightX = Math.max(newFirstTouchPoint.x, newSecondTouchPoint.x);
+      var newTopY = Math.min(newFirstTouchPoint.y, newSecondTouchPoint.y);
+      var newBottomY = Math.max(newFirstTouchPoint.y, newSecondTouchPoint.y);
+
+      var newAvgX = (newLeftX + newRightX) / 2;
+      var newAvgY = (newBottomY + newTopY) / 2;
+      var newDiffX = newRightX - newLeftX;
+      var newDiffY = newBottomY - newTopY;
+
+      if (this._xScale != null && newDiffX !== 0 && oldDiffX !== 0) {
+        this._xScale.domain(PanZoom.magnify(this._xScale, oldDiffX / newDiffX, oldAvgX));
+        this._xScale.domain(PanZoom.translate(this._xScale, oldAvgX - newAvgX));
+      }
+      if (this._yScale != null && newDiffY !== 0 && oldDiffY !== 0) {
+        this._yScale.domain(PanZoom.magnify(this._yScale, oldDiffY / newDiffY, oldAvgY));
+        this._yScale.domain(PanZoom.translate(this._yScale, oldAvgY - newAvgY));
       }
-      this._zoom.on("zoom", () => this._rerenderZoomed());
-      this._zoom(this._hitBox);
     }
 
-    public _anchor(component: Component.AbstractComponent, hitBox: D3.Selection) {
-      super._anchor(component, hitBox);
-      this.resetZoom();
+    private _handleTouchEnd(ids: number[], idToPoint: { [id: number]: Point; }, e: TouchEvent) {
+      ids.forEach((id) => {
+        this._touchIds.remove(id.toString());
+      });
+      if (this._touchIds.size() < 2) {
+        this._setupDragInteraction();
+      }
     }
 
-    public _requiresHitbox() {
-      return true;
+    private static magnify<D>(scale: Scale.AbstractQuantitative<D>, magnifyAmount: number, centerValue: number) {
+      var magnifyTransform = (rangeValue: number) => scale.invert(centerValue - (centerValue - rangeValue) * magnifyAmount);
+      return scale.range().map(magnifyTransform);
     }
 
-    private _rerenderZoomed() {
-      // HACKHACK since the d3.zoom.x modifies d3 scales and not our TS scales, and the TS scales have the
-      // event listener machinery, let's grab the domain out of the d3 scale and pipe it back into the TS scale
-      if (this._xScale) {
-        var xDomain = (<any> this._xScale)._d3Scale.domain();
-        this._xScale.domain(xDomain);
-      }
+    private static translate<D>(scale: Scale.AbstractQuantitative<D>, translateAmount: number) {
```

Comment:
Probably `translateScale` is a better name, since the method isn't on the `Scale`.